mapper-runtime/src/main/java/com/datastax/oss/driver/api/mapper/MapperBuilder.java

```diff
@@ -33,13 +33,17 @@
  */
 public abstract class MapperBuilder<MapperT> {
 
+  public static final String SCHEMA_VALIDATION_ENABLED_SETTING =
+      "datastax.mapper.schemaValidationEnabled";
   protected final CqlSession session;
   protected CqlIdentifier defaultKeyspaceId;
   protected Map<Object, Object> customState;
 
   protected MapperBuilder(CqlSession session) {
     this.session = session;
     this.customState = new HashMap<>();
+    // schema validation is enabled by default
+    customState.put(SCHEMA_VALIDATION_ENABLED_SETTING, true);
```

Comment:
I would have used `false` as the default, but now that I think about it I can see pros and cons on both sides:
* enabled by default: errors are reported sooner. But if users don't realize that they can disable it, the (possibly expensive?) checks will run every time a new DAO gets created.
* disabled by default: more efficient by default. But errors are only reported at usage time, and might be less clear (stack trace involving generated mapper classes). Also they will be detected one at a time.

In the end I think I agree with enabled by default.

Reply:
I agree - I think that the overhead of checking for every `Dao` maybe not so big because it is happening only one time at the compile time.